Common/cpp/LayoutAnimations/LayoutAnimationsProxy.cpp

```diff
@@ -11,35 +11,100 @@ namespace reanimated {
 const long long idOffset = 1e9;
 
 LayoutAnimationsProxy::LayoutAnimationsProxy(
-    std::function<void(int, jsi::Object newProps)> _notifyAboutProgress,
-    std::function<void(int, bool)> _notifyAboutEnd)
-    : notifyAboutProgress(std::move(_notifyAboutProgress)),
-      notifyAboutEnd(std::move(_notifyAboutEnd)) {}
+    std::function<void(int, jsi::Object newProps)> progressHandler,
+    std::function<void(int, bool, std::string)> endHandler)
+    : progressHandler(std::move(progressHandler)),
+      endHandler(std::move(endHandler)) {}
 
 void LayoutAnimationsProxy::startObserving(
     int tag,
     std::shared_ptr<MutableValue> sv,
     jsi::Runtime &rt) {
   observedValues[tag] = sv;
+  this->progressHandler(tag, sv->value->toJSValue(rt).asObject(rt));
   sv->addListener(tag + idOffset, [sv, tag, this, &rt]() {
-    std::shared_ptr<FrozenObject> newValue =
-        ValueWrapper::asFrozenObject(sv->value->valueContainer);
-    this->notifyAboutProgress(tag, newValue->shallowClone(rt));
+    this->progressHandler(tag, sv->value->toJSValue(rt).asObject(rt));
   });
 }
 
-void LayoutAnimationsProxy::stopObserving(int tag, bool finished) {
+void LayoutAnimationsProxy::stopObserving(
+    int tag,
+    bool finished,
```

Comment:
As far as I can tell, the flag here is `finished` instead of `cancel`, since that's what the callbacks on animations are passed - I'm not sure being consistent with either of these two makes more sense than the other.